cubic-server/command_parser/commands/Time.cpp

```diff
@@ -0,0 +1,153 @@
+#include "Time.hpp"
+#include "Player.hpp"
+#include "Server.hpp"
+#include "World.hpp"
+
+using namespace command_parser;
+
+void Time::autocomplete(std::vector<std::string> &args, Player *invoker) const
+{
+    if (invoker)
+        return;
+    else
+        LINFO("autocomplete time");
+}
+
+int setTimeFromArg(std::string value, int multiplier)
+{
+    int time = 0;
+    try {
+        if (value.back() == 'd' || value.back() == 's' || value.back() == 't')
+            value.pop_back();
+        time = std::stod(value) * multiplier;
+        return time;
+    } catch (const std::exception &e) {
+        LERROR("Expected float");
+        // invoker->sendPlayerChatMessage("Expected float");
+        return -1;
+    }
+}
+
+int setMultiplier(char arg)
+{
+    if (arg == 'd') {
+        return 24000;
+    } else if (arg == 's') {
+        return 20;
+    } else if (arg == 't') {
+        return 1;
+    }
+    return -1;
+}
+
+int setTimeToAdd(std::string timeToAdd)
+{
+    int time = 0;
+    if (std::find_if(timeToAdd.begin(), timeToAdd.end(), [](unsigned char c) {
+            return !std::isdigit(c);
+        }) == timeToAdd.end())
+        time = setTimeFromArg(timeToAdd, 1);
+    else {
+        if ((timeToAdd.back() == 'd' || timeToAdd.back() == 's' || timeToAdd.back() == 't') && timeToAdd.front() != '-') {
+            time = setTimeFromArg(timeToAdd, setMultiplier(timeToAdd.back()));
+            return time;
+        } else if (timeToAdd.front() == '-') {
+            LERROR("Tick count must be non-negative");
+            // invoker->sendPlayerChatMessage("Tick count must be non-negative");
+            return -1;
+        }
+        LERROR("Expected float");
+        // invoker->sendPlayerChatMessage("Expected float");
+        return -1;
+    }
+    return time;
+}
+
+int setTimeToSet(std::string timeToSet)
+{
+    int time = 0;
+    if (std::find_if(timeToSet.begin(), timeToSet.end(), [](unsigned char c) {
+            return !std::isdigit(c);
+        }) == timeToSet.end())
+        time = setTimeFromArg(timeToSet, 1);
+    else if (timeToSet == "day")
+        time = 1000;
+    else if (timeToSet == "night")
+        time = 13000;
+    else if (timeToSet == "noon")
+        time = 6000;
+    else if (timeToSet == "midnight")
+        time = 18000;
+    else {
+        if (timeToSet.back() == 'd' || timeToSet.back() == 's' || (timeToSet.back() == 't' && timeToSet != "midnight" && timeToSet != "night") && timeToSet.front() != '-') {
```

Comment:
Why do that ?

Reply:
According to https://minecraft.fandom.com/wiki/Commands/time :
![image](https://user-images.githubusercontent.com/62648027/231279662-35748edd-a35f-42ae-b029-a7fb8f1315e8.png)